plugins/asdf/asdf.plugin.zsh

```diff
@@ -12,8 +12,6 @@ fi
 if [[ -f "$ASDF_DIR/asdf.sh" ]]; then
     . "$ASDF_DIR/asdf.sh"
 
-    # Load completions
-    if [[ -f "$ASDF_COMPLETIONS/asdf.bash" ]]; then
-        . "$ASDF_COMPLETIONS/asdf.bash"
-    fi
+    fpath=($ASDF_COMPLETIONS $fpath)
+    autoload -Uz compinit && compinit
```

Comment:
This shouldn't be performed for any brew installed version of `asdf` as Homebrew has it's own setup for completions, which I believe I implemented in https://github.com/Homebrew/homebrew-core/blob/6e728e63ab37243b6832d2995e7efdbac7f0d612/Formula/asdf.rb#L25 (the correctness of my implementation needs to be validated).

Reply:
I believe #L8 can be removed also as the Homebrew install should now handle the completions.

Reply:
I'm not familiar with brew, so I may be completely wrong here.

The documentation suggest that `FPATH=$(brew --prefix)/share/zsh/site-functions:$FPATH` is needed to install all zsh completions when using brew, not just one plugin. ZSH [brew plugin](https://github.com/ohmyzsh/ohmyzsh/blob/7d6f754b6807e312089179e3a4719145c58c7986/plugins/brew/brew.plugin.zsh) also doesn't do anything related to completions.

I think that your changes on brew formulae for asdf should sync link the _asdf file and as users of brew follow the instructions on https://docs.brew.sh/Shell-Completion#configuring-completions-in-zsh, _it just work_ ®.

Reply:
I made a few changes to not add the completion code on brew, however I can't test it on brew.

Reply:
I have a suspicion that we should also be handling here different versions of asdf. Which asdf version added the zsh completion?

Reply:
I can perform a test of brew tomorrow, though am also not particularly familiar.

`0.7.6` was the last version this plugin properly supports.

`0.7.8` added the ZSH completions properly.